src/neural/loader.cc

```diff
@@ -46,19 +50,83 @@ FloatVectors LoadFloatsFromFile(const std::string& filename) {
     } else {
       bytes_read += sz;
       buffer.resize(bytes_read);
-      // Add newline in the end for the case it was not there.
-      buffer.push_back('\n');
       break;
     }
   }
   gzclose(file);
 
+  return buffer;
+}
+
+void DenormLayer(const lc0::Weights_Layer &layer, FloatVectors &vecs) {
+  FloatVector v;
+  auto &buffer = layer.params();
+  auto data = reinterpret_cast<const std::uint16_t *>(buffer.data());
+  int n = buffer.length() / 2;
+  v.resize(n);
+  for (int i = 0; i < n; i++) {
+    v[i] = data[i] / float(0xffff);
+    v[i] *= layer.max_val() - layer.min_val();
+    v[i] += layer.min_val();
+  }
+  vecs.emplace_back(v);
+}
+
+void DenormConvBlock(const lc0::Weights_ConvBlock &conv, FloatVectors &vecs) {
+  DenormLayer(conv.bn_stddivs(), vecs);
```

Comment:
Shouldn't this order be reversed?  PopulateConvBlockWeights expects to pop off the end in this order, so we should pop them on in reverse I think.

Reply:
No :)